Landlord submission form

brentonmdunn opened the pull request:
## Tracking Info

Resolves #17

## Changes

<!-- What changes did you make? -->

- Add all components for form
- Performed validation on input

## Testing

<!-- How did you confirm your changes worked? -->

- Tested all of the inputs to validate that the radio and checkboxes worked.

## Confirmation of Change

<!-- Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change. -->

<img width="1512" alt="image" src="https://github.com/TritonSE/USHS-Housing-Portal/assets/47510212/8aced4d4-fcc1-4ada-a600-a112cc06d45a">

<img width="1512" alt="image" src="https://github.com/TritonSE/USHS-Housing-Portal/assets/47510212/d20901ab-3e25-448c-a206-8a72686ffea5">

<img width="1512" alt="image" src="https://github.com/TritonSE/USHS-Housing-Portal/assets/47510212/ac3d7b6d-4285-476c-acd9-3fb3585db097">

<img width="1512" alt="image" src="https://github.com/TritonSE/USHS-Housing-Portal/assets/47510212/b92766a8-8cae-4d62-a31a-05d2041bcd3e">

<img width="1512" alt="image" src="https://github.com/TritonSE/USHS-Housing-Portal/assets/47510212/9eaac77a-3b8d-4371-8998-4ca86a787521">

brentonmdunn committed:
Merged main into branch

brentonmdunn committed:
Added all components, useStates, and handlers

brentonmdunn committed:
Add custom styles to input checkbox/radio

brentonmdunn committed:
Fix most styles

brentonmdunn committed:
Lint fixes and consistency

brentonmdunn committed:
Add logo

petabite reviewed (changes requested):
Nice work! 💯 This is looking really solid so far!

I demoed this during the team meeting and we noticed a few things:

- [x] Can we make the "Apartment/Suite" field not mandatory?
- [x] Increase spacing between the text box and n/a for the "Application Fee Cost" field
- [x] The "Additional fields to be entered by housing locator" section heading and label text should be a different color to highlight to HLs that they need to fill it out (see designs)
- [x] The "Additional Rules/Regulations" section has a bug where selecting some options and then selecting N/A does nothing but if you click it again, it fills in the radio button.
- [x] The Additional comments text box for the landlord is missing
- [x] Was not in the designs before, but can we turn the two comments text boxes (for landlord and HL) into textareas since it is possible they could could be typing in a lot of text?
- [x] The various number inputs (e.g. application fee cost, phone number, rent per month, security deposit, etc) should only allow numbers to be inputed. Currently, typing in letters is allowed
- [x] Was not in the designs before: For the "Select All Appliances that Apply or N/A" heading, can we delete the "or N/A" part?
- [x] Was not in the designs before: For "Payment/Renting Criteria" can we remove the “(select ALL that apply)” text
- [x] Is it possible to max the year part of the Date Available input to 4 digits (i.e. set the max year to 9999 or something)?
- [x] It looks like there are a few type errors in the `src/components/ListingFormComponents.tsx` file

I know this is a lot 😓  Please let me know if you have any questions about anything. Thanks!

brentonmdunn committed:
PR changes

brentonmdunn committed:
Unsuccessful type fix

soma-p committed:
Validators Complete

soma-p committed:
combining my edits with brenton's
Merge branch 'landlord-submission-form' of https://github.com/TritonSE/USHS-Housing-Portal into landlord-submission-form

brentonmdunn committed:
Fix eslint errors

brentonmdunn committed:
Make edits to styles

brentonmdunn committed:
Fix typo, change type names for formType

soma-p committed:
Finished OnSubmit

soma-p committed:
Finished Validation and Other Cases

soma-p committed:
Resolved Merge Conflicts

soma-p committed:
Fixed validation for submission

soma-p commented:
Added validation + onSubmit handler to the Landlord Form. Tested by running the front-end. Pending items: code for the building of the initial and final forms. Password code completed, to be added once initial and final forms HTML and styled components are complete.

YashRavipati1 commented:
Not sure if others are facing the same issue or if this is just because the branch has not been merged with main in a while, but I can't seem to properly login to check the actual page. Everytime I try to login it just redirects me to the login page again

YashRavipati1 commented:
> Not sure if others are facing the same issue or if this is just because the branch has not been merged with main in a while, but I can't seem to properly login to check the actual page. Everytime I try to login it just redirects me to the login page again

So it seems to be trying to create a new user on the staging environment everytime I try to login. Could be something to look into?

YashRavipati1 commented on frontend/src/App.tsx:
```diff
@@ -6,24 +6,34 @@ import { ThemeProvider } from "styled-components";
 
 function AppRouter() {
   const { signedIn } = useContext(AuthContext);
+  console.log(signedIn);
```
Make sure to remove this

YashRavipati1 commented on frontend/src/contexts/AuthContext.tsx:
```diff
@@ -35,10 +35,7 @@ export function AuthProvider({ children }: ProviderProps) {
-    () => ({ currentUser, signedIn: currentUser !== undefined, loading }),
-    [currentUser],
-  );
+  const value = useMemo(() => ({ currentUser, signedIn: !currentUser, loading }), [currentUser]);
```
Think the issue with sign in could be here, as I believe you're setting signedIn to true if the currentUser doesn't exist and false if it does.

petabite replied:
```suggestion
  const value = useMemo(() => ({ currentUser, signedIn: !!currentUser, loading }), [currentUser]);
```

Ah good catch. This should fix it

YashRavipati1 reviewed (commented):
Quick change to auth context to ensure we can actually see the pages. I can test by just making the change myself as well, but thought you should be aware of the issue.

brentonmdunn committed:
Remove console.logs

brentonmdunn committed:
Fix typo

brentonmdunn committed:
Fix NA bug

brentonmdunn committed:
Fix lint errors

brentonmdunn committed:
Yash/Philip PR comment fix

petabite committed:
Merge remote-tracking branch 'origin/main' into landlord-submission-form

petabite committed:
working form + validation error message

petabite committed:
lint

petabite committed:
update logos

petabite committed:
working password flow

petabite committed:
password in env var

petabite committed:
Merge remote-tracking branch 'origin/main' into landlord-submission-form

petabite committed:
fix firebase json

petabite committed:
pin firebaseToolsVersion

petabite committed:
VITE_APP_LANDLORD_FORM_PASSWORD in ci

petabite committed:
cleanup

petabite reviewed (approved):
🚀
